python/pyspark/sql/pandas/serializers.py

```diff
@@ -194,36 +191,61 @@ def create_array(s, dt: DataType, t: pa.DataType):
                     raise e
             return array
 
-        arrs = []
-        for s, dt in series:
-            t = to_arrow_type(dt) if isinstance(dt, DataType) else dt
-            if t is not None and pa.types.is_struct(t) and not isinstance(dt, UserDefinedType):
-                if not isinstance(s, pd.DataFrame):
-                    raise ValueError("A field of type StructType expects a pandas.DataFrame, "
-                                     "but got: %s" % str(type(s)))
+        def create_arrs_names(s, t: pa.DataType, dt: DataType = None):
+            # Input partition and result pandas.DataFrame empty, make empty Arrays with struct
```

Comment:
Nit: `Input partition and result pandas.DataFrame empty` <- could you rephrase this setence?